dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java

```diff
@@ -0,0 +1,398 @@
+        log.warn("The relationshipType's left label is: " + relationshipType.getLeftLabel());
+        log.warn("The relationshipType's right label is: " + relationshipType.getRightLabel());
+        log.warn("The relationshipType's left entityType label is: " + relationshipType.getLeftType().getLabel());
+        log.warn("The relationshipType's right entityType label is: " + relationshipType.getRightType().getLabel());
+        log.warn("The relationshipType's left min cardinality is: " + relationshipType.getLeftMinCardinality());
+        log.warn("The relationshipType's left max cardinality is: " + relationshipType.getLeftMaxCardinality());
+        log.warn("The relationshipType's right min cardinality is: " + relationshipType.getRightMinCardinality());
+        log.warn("The relationshipType's right max cardinality is: " + relationshipType.getRightMaxCardinality());
+    }
+
+    private boolean verifyMaxCardinality(Context context, Item itemToProcess,
+                                         int maxCardinality, RelationshipType relationshipType) throws SQLException {
+        List<Relationship> rightRelationships = findByItemAndRelationshipType(context, itemToProcess, relationshipType,
+                                                                              false);
+        if (rightRelationships.size() >= maxCardinality && maxCardinality != 0) {
+            return false;
+        }
+        return true;
+    }
+
+    private boolean verifyEntityTypes(Item itemToProcess, EntityType entityTypeToProcess) {
+        List<MetadataValue> list = itemService.getMetadata(itemToProcess, "relationship", "type", null, Item.ANY);
+        if (list.isEmpty()) {
+            return false;
+        }
+        String leftEntityType = list.get(0).getValue();
+        if (!StringUtils.equals(leftEntityType, entityTypeToProcess.getLabel())) {
+            return false;
+        }
+        return true;
+    }
+
+    public Relationship find(Context context, int id) throws SQLException {
+        Relationship relationship = relationshipDAO.findByID(context, Relationship.class, id);
+        return relationship;
+    }
+
+    @Override
+    public List<Relationship> findByItem(Context context, Item item) throws SQLException {
+
+        List<Relationship> list = relationshipDAO.findByItem(context, item);
+
+        list.sort((o1, o2) -> {
+            int relationshipType = o1.getRelationshipType().getLeftLabel()
+                                     .compareTo(o2.getRelationshipType().getLeftLabel());
+            if (relationshipType != 0) {
+                return relationshipType;
+            } else {
+                if (o1.getLeftItem() == item) {
+                    return o1.getLeftPlace() - o2.getLeftPlace();
+                } else {
+                    return o1.getRightPlace() - o2.getRightPlace();
+                }
+            }
+        });
+        return list;
```

Comment:
This is a situation which was not included in the list of use cases before the configurable entities implementation phase.
It can be verified it there's a use case for this, and what the expected behavior should be.
From an implementation point of view, this is quite easy to adjust. So if this is considered a relevant use case in the next phase, it would be a minor change (it's not related to this part of the codebase though)

Reply:
I checked that in the REST API the item details are not exposed under the relationships endpoint or when relationships are embedded on an item (e.g. https://dspace7-entities.atmire.com/rest/#https://dspace7-entities.atmire.com/rest/api/core/items/f2235aa6-6fe7-4174-a690-598b72dd8e44). So currently there is no security issue.

However, I do think it makes sense that when retrieving all publications of an author DSpace would only return publications to which you actually have access (similar to discovery search). But I agree that this can be added as an improvement in another PR.